### PR TITLE
Create intake item

### PR DESCRIPTION
**Pull request by internettrans**
UX needs to be polished

image

### CODE REVIEW

File: frontend/Nouns/CreateEditIntakeForm.test.tsx [hunk 1]
```diff
@@ -1,39 +1,108 @@
...
+            path="/create-intake-form/:nounId"
             component={CreateEditIntakeForm}
           />
         </MemoryRouter>
       </QueryClientProvider>
     );
 
-    expect(await w.findByLabelText(`First Name`)).toBeInTheDocument();
-    expect(await w.findByLabelText(`Last Name`)).toBeInTheDocument();
+    await waitFor(() =>
```
Comment (rhiannonstanford) [inline frontend/Nouns/CreateEditIntakeForm.test.tsx:92]: I like the way this is written.  Very clear to read and understand.

File: frontend/Nouns/CreateEditIntakeForm.tsx [hunk 2]
```diff
@@ -202,6 +264,30 @@ function reducer(state: State, action: Action): State {
         ...state,
         itemToEdit: undefined,
       };
+    case ActionTypes.CreateItem:
```
Comment (rhiannonstanford) [inline frontend/Nouns/CreateEditIntakeForm.tsx:267]: It was here in the code that I began to wonder if creating a separate file or directory for actions or reducers was needed or appropriate.

File: frontend/Nouns/EditIntakeField/EditIntakeTextField.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { Toggle } from "../../Styleguide/Toggle";
+import { Button, ButtonKind } from "../../Styleguide/Button";
+import { Field } from "../../../backend/DB/models/field";
+import { Select } from "../../Styleguide/Select";
+
+export function EditIntakeTextField(props: EditItemProps) {
+  const intakeItem = props.intakeItem as IntakeFieldItem;
+  const [label, setLabel] = useState(intakeItem?.question.label ?? "");
+  const [required, setRequired] = useState(
+    intakeItem?.question.required ?? true
```
Comment (rhiannonstanford) [inline frontend/Nouns/EditIntakeField/EditIntakeTextField.tsx:17]: The left side of this optional boolean can yield the value of null, true, or false.  This boolean will result in either true  or false.   "Nil coalescing operator"

**Review by rhiannonstanford — APPROVED** [review]
Great work here.  Just left a few comments in consideration of organizing the actions and reducers.  Thank you!

File: frontend/Nouns/CreateEditIntakeForm.tsx (hunk 2 above)
Comment (internettrans) [reply frontend/Nouns/CreateEditIntakeForm.tsx:267]: This file is definitely getting pretty big, but some of that is due to sample data and api calls being inlined into this component.

I generally don't like the redux pattern of creating `reducer.js`, `actions.js`, `constants.js` files, as I prefer to colocate the logic for the component within the same file as the component. One of the parts that I don't like about redux is that you have to open up 3-4 files just to see what happens when you click on something.

Redux reducers are for global state and so they're reused across components, but `useReducer` is specific to one component, so reusability is generally not a concern. So I usually just put the logic for it into the same file as the compoennt

**Pull request merged**